integrationtest/org/daisy/dotify/formatter/test/resource-files/flow-in/flow-in-footer-with-marker-reference-expected.pef

```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<pef version="2008-1" xmlns="http://www.daisy.org/ns/2008/pef">
+<pef version="2008-1" xmlns="http://www.daisy.org/ns/2008/pef" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
```

Comment:
Where does this come from? I know it's also in the source OBFL, but can't figure out what change in the code triggered this change in the PEF. Not wrong, just not so nice because the namespace isn't used in the PEF.

Reply:
Oh I see you are injecting fake `MetaDataItem` in the OBFL parser. Hmm I wonder if there is a better way to do it...

Reply:
The trick here I think is to collect all the used namespaces (in the `external-reference`, you should be able to do that with `StartElement.getNamespaceContext()`), then, for each URI select one prefix (there could be more than one in theory), then use this info in `PEFMediaWriter` to create the namespace nodes on the `pef` element, and then when adding the attributes on `row` look up the right prefix in these namespace bindings.

Reply:
You might still have to use the `getMetaData()` trick to pass the info from the parser to the writer. I think it's really a hack but it might be fine if documented properly. Another solution could be to collect the namespace bindings on the writer side, but that means you can only start writing to the output stream when all the rows have been provided, which means it would not really be "streaming" anymore strictly speaking, don't think that's really a problem though. Of course another solution is to not put the namespace nodes on `pef` but on `volume`, `section`, `page` or `row`, which might result in a lot of them, but that's also not really a problem.

Reply:
I would rather not put the namespaces on each row because that would be a lot of unnecessary overhead. And the hack to get the metadata from one document to the other seemed like a good compromise. Then again, I don't know if I documented it well enough. 

We could create a very sophisticated solution for this extra feature, or we could keep it light.
My suggestion is to try to document that when you need extra fields in a different namespace, you put that namespace on the root of the document to get them propagated. 

That would work with my solution and keep it simple.

Using a real XML writer for the PEFWriter in the future could be an excellent addition and changing this behavior than could be an option. I could add more if you like, but I don't want to get too far of scope for this solution.

Reply:
OK fair enough, we can keep the getMetadata trick to make it easier on the writer side, but it's really a total hack in my opinion, for several reasons, so needs to be documented very well. However on the parser side there is really nothing stopping us from parsing the XML correctly.

Reply:
The extra namespace should now be removed and I've implemented the meta fetch in the parseExternalReference instead.